src/bootstrap/native.rs

```diff
@@ -91,6 +91,64 @@ pub fn prebuilt_llvm_config(
     Err(Meta { stamp, build_llvm_config, out_dir, root: root.into() })
 }
 
+// modified from `check_submodule` and `update_submodule` in bootstrap.py
+fn update_llvm_submodule(builder: &Builder<'_>) {
+    let llvm_project = &Path::new("src").join("llvm-project");
+
+    // check_submodule
+    let checked_out = if builder.config.fast_submodules {
+        Some(output(
+            Command::new("git")
+                .args(&["rev-parse", "HEAD"])
+                .current_dir(builder.config.src.join(llvm_project)),
+        ))
+    } else {
+        None
+    };
+
+    // update_submodules
+    let recorded = output(
+        Command::new("git")
+            .args(&["ls-tree", "HEAD"])
+            .arg(llvm_project)
+            .current_dir(&builder.config.src),
+    );
+    let hash = recorded.split(' ').nth(2).unwrap();
+
+    // update_submodule
+    if let Some(llvm_hash) = checked_out {
+        if hash == llvm_hash {
+            // already checked out
+            return;
+        }
+    }
+
+    println!("Updating submodule {}", llvm_project.display());
+    builder.run(
+        Command::new("git")
+            .args(&["submodule", "-q", "sync"])
+            .arg(llvm_project)
+            .current_dir(&builder.config.src),
+    );
+    builder.run(
+        Command::new("git")
+            .args(&["submodule", "update", "--init", "--recursive", "--progress"])
```

Comment:
Note that before bootstrap.py did version detection to see if `--progress` was available; given that git 2.11 is pretty common (2.25 is packaged on ubuntu, 2.27 is packaged on centos) and there are so many ways to avoid building LLVM, I don't think it's worth the maintenance burden to port it over.

Reply:
Can you track down the addition of that check and verify that the reasons we added it no longer hold true?

Reply:
Here's all the history for it:

$ git log --oneline -S '--progress' ../bootstrap/bootstrap.py
7ac093fda9f Detect git version before attempting to use --progress
a0d63354f23 intelligently handle older version of git in bootstrap
49eb1e54199 x.py: fixup 6130fc884bc1dff9bb835894a7bb2042c110b011
6130fc884bc Add --progress to git submodule commands

The first time it was removed was in 49eb1e54199 (https://github.com/rust-lang/rust/pull/57082), which says it fixes https://github.com/rust-lang/rust/issues/57080. The most detailed comment I saw there was https://github.com/rust-lang/rust/issues/57080#issuecomment-460546880:

> I'm still on Ubuntu Xenial, which is still in service until 2021 and ships Git 2.7.4. I wonder if would it be too petty if I sent a PR to make bootstrap.py check the Git version in order to decide whether to pass --progress? I think it's convenient to be able to hack on rustc with a moderately old Git version.

Xenial is supported until April 2021 (https://wiki.ubuntu.com/Releases), which is in 2 months. If you like I can remove `--progress` for now and add it back in April? I don't think it's worth messing with semver parsing when it's being deprecated so soon.

Reply:
Can we run it and then if it errors (for any reason) rerun it without --progress?

Reply:
Sure thing, done.